### PR TITLE
CB-10075 Upgrade image selection logic should not depend on image catalog URL

### PR DESCRIPTION
**Pull request by schfeca75**
CB-10075 Upgrade image selection logic should not depend on image catalog URL

* First try to use imagecatalog by name given in the current image
* if image catalog is not found by name, use the stored imagecatalog url instead

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java [hunk 1]
```diff
@@ -180,7 +185,14 @@ private Image getCurrentImageFromCatalog(String currentImageId, CloudbreakImageC
         return imageService.getImage(stack.getId());
     }
 
-    private CloudbreakImageCatalogV3 getImagesFromCatalog(String imageCatalogUrl) throws CloudbreakImageCatalogException {
+    private CloudbreakImageCatalogV3 getImagesFromCatalog(Workspace workspace,
+            com.sequenceiq.cloudbreak.cloud.model.Image image) throws CloudbreakImageCatalogException {
+        String imageCatalogUrl;
+        try {
```
Comment (keyki) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java:191]: Perhaps you could extend these branches with log messages why we are using a certain url.

File: core/src/test/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityServiceTest.java [hunk 2]
```diff
@@ -155,18 +167,56 @@ public void testCheckForUpgradesByNameShouldReturnsImagesWhenThereAreAvailableIm
                 currentImage.getImageCatalogName());
     }
 
+    @Test
+    public void testCheckForUpgradesByNameShouldReturnsImagesFromFallbackCatalogWhenThereAreAvailableImagesButImageCatalogByNameNotFound()
```
Comment (pdarvasi) [inline core/src/test/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityServiceTest.java:171]: typo: returns -> return

Comment (pdarvasi) [inline core/src/test/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityServiceTest.java:171]: could you pls have a unit test to cover "ImageCatalogByNameFound()"

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java (hunk 1 above)
Comment (pdarvasi) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java:191]: +1

File: core/src/test/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityServiceTest.java (hunk 2 above)
Comment (keyki) [reply core/src/test/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityServiceTest.java:171]: omg

**Review by keyki — APPROVED** [review]

File: core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java [hunk 3]
```diff
@@ -180,7 +185,18 @@ private Image getCurrentImageFromCatalog(String currentImageId, CloudbreakImageC
...
-    private CloudbreakImageCatalogV3 getImagesFromCatalog(String imageCatalogUrl) throws CloudbreakImageCatalogException {
+    private CloudbreakImageCatalogV3 getImagesFromCatalog(Workspace workspace,
+            com.sequenceiq.cloudbreak.cloud.model.Image image) throws CloudbreakImageCatalogException {
+        String imageCatalogName = image.getImageCatalogName();
+        String imageCatalogUrl;
+        try {
+            imageCatalogUrl = imageCatalogService.get(workspace.getId(), imageCatalogName).getImageCatalogUrl();
+            LOGGER.info("Image catalog with name {} and url {} is used for image filtering.", imageCatalogName, imageCatalogUrl);
+        } catch (NotFoundException ex) {
+            imageCatalogUrl = image.getImageCatalogUrl();
```
Comment (lacikaaa) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java:196]: if you declare `imageCatalogUrl` on line 191 with this as a default value, then you can omit this

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**